fast64_internal/sm64/sm64_level_writer.py

```diff
@@ -482,26 +513,18 @@ def stringToMacros(data):
         arguments = arguments.split(",")
         for i in range(len(arguments)):
             arguments[i] = arguments[i].strip()
-
-        macroData.append([macro, arguments, comment])
+        macroData.append(Macro(function, arguments, comment))
 
     return macroData
 
 
-def macroToString(macroCmd, useComma):
-    result = macroCmd[0] + "("
-    for arg in macroCmd[1]:
-        result += arg + ", "
-    result = result[:-2] + ")" + ("," if useComma else "")
-    result += " " + macroCmd[2]
-    return result
+def macroToString(macro_cmd):
+    return f"{macro_cmd.function}({', '.join(macro_cmd.args)}), {macro_cmd.comment}"
```

Comment:
Adding a comma here always causes the game to not compile, DEFINE_LEVEL should not have a comma

Reply:
This is a major issue that you should address first